### PR TITLE
feat: add force login component to design system

### LINKED ISSUE #722: Feature: Force login component
### Type of feature

🍕 Feature

### Current behavior

We need a generic component to apply to any component that blurs the feature and ask for a login. This only needs to be added to the design system. We will add this as an experiment in the future. 

### Suggested solution

The example is the contribution scatter chart, but this component should be flexible enough to blur any child component. 

Screen Shot 2022-12-22 at 11 37 14 AM

[[Figma](https://www.figma.com/file/OpVX6WT7dmWqnwRuEvADMF/Design-Lab?type=design&node-id=441-874&t=CSlobWSNVftBnjsN-0)]

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [ ] I agree to follow this project's Contribution Docs

### PR DESCRIPTION
**Pull request by babblebey**
## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [x] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Description

This PR adds a component that forces login (i.e. restrict child content access to only users who are logged in) to the design system. It addresses issue #722 which states the need for  a component that performs this function.

The Component with draft name `AuthContentWrapper` (open to better name suggestion) is flexible such that it creates a blur effect on any child component it receives. Below is a brief description of how it works

- The component works by creating a wrapper around child component it seeks to blur
```javascript
<AuthContentWrapper>
  <ChildComponent />
</AuthContentWrapper>
```
- A blur is created conditional based on current user's authentication status i.e. child component is blurred when user auth status is `false` and not blurred when auth status is `true`
- The component uses the `useSupabaseAuth` hook's `user` object to determine the user authentication status
- The component implements the `Button` component and the `useSupabaseAuth` hook's `signIn` function to create a `Connect with GitHub` which encourage user to login to remove restriction from blurred content.

### **Code Implementation Sample**

_**With Dashboard Chart as Child Component**_

```javascript
<div className="flex flex-col w-full">
  <AuthContentWrapper>
    <Card className="w-full">
      <NivoScatterPlot
        title="Contributor Distribution"
        ...
      />
    </Card>
  </AuthContentWrapper>
</div>
```

**Result**

screenshot-localhost_3000-2023 07 04-18_36_42

_**With Dashboard Highlight Cards Section as Child Component**_

```javascript
<AuthContentWrapper>
  <section className="...">
    <HighlightCard
      label="Contributors"
      ...
    />
    <HighlightCard
      label="Spam"
      ...
    />
    <HighlightCard
      label="Accepted PRs"
     ...
    />
    <HighlightCard
      label="Unlabeled PRs"
      ...
    />
  </section>
</AuthContentWrapper>
```

**Result**

screenshot-localhost_3000-2023 07 04-20_15_58

_**With Dashboard Repositories Tool as Child Component**_

```javascript
<AuthContentWrapper>
  <Repositories repositories={repositories} />
</AuthContentWrapper>
```

**Result**

screencast-localhost_3000-2023 07 04-20_31_25

## Related Tickets & Documents

Fixes #722 

## Mobile & Desktop Screenshots/Recordings

**Authenticated (Logged In) User Case**

screenshot-localhost_3000-2023 07 04-18_53_10

**Unauthenticated (Logged Out) User Case - On Disconnect/Logout Action**

screencast-localhost_3000-2023 07 04-19_04_18

screenshot-localhost_3000-2023 07 04-19_49_23 (1)

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [x] 📕 storybook
- [ ] 🙅 no documentation needed

## [optional] Are there any post-deployment tasks we need to perform?
NA

## [optional] What gif best describes this PR or how it makes you feel?
_I need GIF pluggggg please_ 🥺

### CODE REVIEW

File: components/molecules/AuthContentWrapper/auth-content-wrapper.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import React from "react";
+import clsx from "clsx";
+
+interface AuthContentWrapperProps {
+  isAuthenticated: boolean;
+  children: React.ReactNode;
+}
+
+const AuthContentWrapper: React.FC<AuthContentWrapperProps> = ({ isAuthenticated, children }) => {
```
Comment (babblebey) [inline components/molecules/AuthContentWrapper/auth-content-wrapper.tsx]: Kindly note, the name is very much a Draft!

Unless y'all think this is just fine.

**Comment by babblebey** [comment]
Kindly note, the name `AuthContentWrapper` is very much a Draft!

I'm open to better name suggestion, unless y'all think this is just fine.

**Review by bdougie — APPROVED** [review]
The implementation and naming I will leave for @OgDev-01 or @brandonroberts to review. 

The parent/children implement is exactly what I had in mind. 

cc @isabensusan as FYI that we may want to start using this component in future designs for the reports and other paid features.

**Review by brandonroberts — APPROVED** [review]

File: components/molecules/AuthContentWrapper/auth-content-wrapper.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  useEffect(() => {
+    if (typeof window !== "undefined") {
+      setHost(window.location.origin as string);
+    }
+  }, []);
+
+  return (
+    <div className="relative">
+      {!isAuthed && (
+        <div className="absolute z-10 w-full h-full top-0 left-0 flex items-center justify-center rounded-lg bg-white/50">
```
Comment (NsdHSO) [inline components/molecules/AuthContentWrapper/auth-content-wrapper.tsx:32]:
```suggestion
        <div className="absolute z-10 w-full h-full flex items-center justify-center rounded-lg bg-white/50">
```
Do we need these classes??

File: components/molecules/AuthContentWrapper/auth-content-wrapper.tsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  children: React.ReactNode;
+}
+
+const AuthContentWrapper: React.FC<AuthContentWrapperProps> = ({ children }) => {
+  const router = useRouter();
+  const currentPath = router.asPath;
+
+  const { signIn, user: isAuthed } = useSupabaseAuth();
+
+  const [host, setHost] = useState<string>("");
```
Comment (OgDev-01) [inline components/molecules/AuthContentWrapper/auth-content-wrapper.tsx:22]: nit: the type is inferred automatically in this case
```suggestion
  const [host, setHost] = useState("");
```

**Review by OgDev-01 — APPROVED** [review]
LGTM 🍕

File: components/molecules/AuthContentWrapper/auth-content-wrapper.tsx (hunk 2 above)
Comment (babblebey) [reply components/molecules/AuthContentWrapper/auth-content-wrapper.tsx:32]: Yes, they are imperative to set the location of the absolutely positioned element.

Without them, the element isn't positioned properly to the top and left of its relative parent element.

Comment (NsdHSO) [reply components/molecules/AuthContentWrapper/auth-content-wrapper.tsx:32]: @babblebey IMHO, I think you set the display at your division `flex` and you use justify-center and items-center.

Comment (babblebey) [reply components/molecules/AuthContentWrapper/auth-content-wrapper.tsx:32]: Very well.

But `flex` and it's related properties `items-center` & `justify-center` are `display` properties for the element's child items i.e. the `Button` component, they determine how the component is rendered in box-types.

We require positioning for the element itself, hence the `absolute`, this allows us set "how" and "where" the element is positioned on the entire web page. 

The "how" is the `absolute` while the "where" are the `top` and `left. They work exactly together to create a crisp positioning.

In the absence of "where", we're living the location for the browser to decide, and this in most cases doesn't position the element well.

**Pull request merged**